common/workunit/workunit.cpp

```diff
@@ -9397,6 +9827,112 @@ IWUGraphStats *CLocalWorkUnit::updateStats(const char *graphName, StatisticCreat
     return new CWuGraphStats(LINK(p), creatorType, creator, _wfid, graphName, subgraph);
 }
 
+bool CLocalWorkUnit::usingDedicatedLogFiles(const char *processType, const char *processName, const char *logSpec) const
+{
+    if (isEmptyString(processType) || (isEmptyString(processName) && isEmptyString(logSpec)))
+        return false;
+
+    IArrayOf<IConstWUProcessLogInfo> processLogs;
+    if (!isEmptyString(processName))
+        getWUProcessLogInfo(processType, processName, processLogs);
+    else
+    {
+        StringBuffer path, logName, process;
+        splitFilename(logSpec, NULL, &path, &logName, NULL);
+
+        const char *searchStr = strieq(processType, "Thor") ? THORMASTERLOGSEARCHSTR : ECLAGENTLOGSEARCHSTR;
+        if (strncmp(logName, searchStr, strlen(searchStr)) != 0) //example: mythor_thormaster 
+        {// This is an imported WU. Find processName from logName.
+            VStringBuffer afterStr("_%s", searchStr);
+            const char *ptr = logName.str();
+            const char *pptr = strstr(ptr, afterStr);
+            if (!pptr)
+                return false; //Should not happen
+            process.append(pptr - ptr, ptr);
+        }
+        else if (!path.isEmpty())
+        {// Find processName from path
+            path.setLength(path.length() - 1); //Remove the last char which is a PATHSEPCHAR
+            const char *ptr = strrchr(path.str(), PATHSEPCHAR);
+            if (!ptr)
+                return false; //Should not happen
+            process.set(ptr + 1);
+        }
+        else
+            return false; //Should not happen
+
+        getWUProcessLogInfo(processType, process, processLogs);
+    }
+    if (processLogs.ordinality() == 0)
+        return false;
+
+    return processLogs.item(0).getSingleLog();
+}
+
+unsigned CLocalWorkUnit::getNumberOfThorSlaves(const char *processName) const
+{
+    Owned<IPropertyTreeIterator> procs = getProcesses("Thor", processName);
+    if (procs->first())
+    {
+        unsigned numberOfThorSlaves = procs->query().getPropInt("@max", 0);
+        if (numberOfThorSlaves > 0)
+            return numberOfThorSlaves;
+    }
```

Comment:
is it correct to continue here, if there is no the request "Thor" processName is not part of the workunit?

Reply:
Could this happen when the addProcess() has not be called? This is in getNumberOfThorSlaves() which will be removed.